arangod/RocksDBEngine/ReplicatedRocksDBTransactionState.cpp

```diff
@@ -173,6 +175,7 @@ std::lock_guard<std::mutex> ReplicatedRocksDBTransactionState::lockCommit() {
 
 /// @brief abort and rollback a transaction
 Result ReplicatedRocksDBTransactionState::doAbort() {
+  std::shared_lock lock{_collectionsLock};
```

Comment:
Suggestion: move the lock acquisition into the `if (!mustBeReplicated())` branch.
This avoids acquiring the lock unnecessarily, and also makes it clearer for the reader that the lock only protects `_collections`, but not the mutation of `_hasActiveTrx`.

Reply:
Same potential deadlock problem exists here as above.